### ISSUE TITLE
Cannot assign issue to a user using Set-JiraIssue

### ISSUE THREAD

**Issue by michalporeba** [AtlassianPS/JiraPS#320]
## Description

Assigning JIRA issues to a user with Set-JiraIssue suddenly stopped working. 

## Steps To Reproduce

1. start a new session with `New-JiraSession`
2. Execute
```powershell
 (Get-JiraIssue Issue-123) | Set-JiraIssue -Assignee (Get-JiraUser)
```
3. The error is 
```
Invoke-JiraMethod : Can not deserialize instance of java.lang.String out of START_ARRAY token
 at [Source: org.apache.catalina.connector.CoyoteInputStream@184681e6; line: 2, column: 5] (through reference chain: com.atlassian.jira.rest.v2.issue.UserBean["name"])
At C:\Program Files\WindowsPowerShell\Modules\jiraps\2.8.6\Public\Set-JiraIssue.ps1:206 char:21
+                     Invoke-JiraMethod @parameter
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidResult: (:) [Invoke-JiraMethod], RuntimeException
    + FullyQualifiedErrorId : InvalidResponse.Status400,Invoke-JiraMethod
```

## Expected behavior

Up to recently this was assigning the issue to the signed user. 
Changing (Get-JiraUser) to a username string doesn't solve it. 

## Your Environment

JiraPS 2.8.6
PSVersion 5.1.16299.785

**Comment by alexsuslin** [comment]
Get-JiraUser having validation attribute [ValidateNotNullOrEmpty()]
Weird but there is no any error when running this command without parameters

**Comment by lipkau** [comment]
1. Running the command without parameters should return the information about the user that is logged on. As defined in the [docs](https://atlassianps.org/docs/JiraPS/commands/Get-JiraUser/#self-default).

2. The output of `Get-JiraUser` is currently not suitable for `Set-JiraIssue`. This was already reported in #200 .
You can add a `(Get-JiraUser).Name` to work around it for now

**Comment by michalporeba** [comment]
I don't think this is a duplicat @lipkau. I have included `(Get-JiraUser)` in the example so I don't have to show my username, but it fails with the same exception regardless if you use `-Assignee (Get-JiraUser)`, `-Assignee (Get-JiraUser).Name` or if you simply pass a string representing the user like so `-Assignee myusername`

**Comment by lipkau** [comment]
I am unable to replicate the behavior you described:

![image](https://user-images.githubusercontent.com/4992870/49687970-7a4ed980-fb0b-11e8-9570-fc059cfbbce5.png)

**Comment by lipkau** [comment]
Please show how you are running these commands and the output of them.

**Comment by derekbartram** [comment]
I too get this problem; if I run the commands as you have lipkau, I get `Invoke-JiraMethod : Can not deserialize instance of java.lang.String out of START_ARRAY token
 at [Source: org.apache.catalina.connector.CoyoteInputStream@2dd22d10; line: 2, column: 5] (through reference chain: com.atlassian.jira.rest.v2.issue.UserBean["name"])
At C:\Program Files\WindowsPowerShell\Modules\JiraPS\2.9.0\JiraPS.psm1:5335 char:21
+                     Invoke-JiraMethod @parameter
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidResult: (:) [Invoke-JiraMethod], RuntimeException
    + FullyQualifiedErrorId : InvalidResponse.Status400,Invoke-JiraMethod`

**Comment by lipkau** [comment]
Please show me the payload of `$Body` and `$splatParameters | Out-String` when you run the command with `-debug` when you see this message:
https://github.com/AtlassianPS/JiraPS/blob/a0e29a616fda283531d8d127dc031dc2efd9b5fd/JiraPS/Public/Invoke-JiraMethod.ps1#L157

**Comment by michalporeba** [comment]
![image](https://user-images.githubusercontent.com/120880/50829250-ba3bf300-133b-11e9-933b-a6da4728c704.png)
![image](https://user-images.githubusercontent.com/120880/50829259-c58f1e80-133b-11e9-93eb-b13211b005e3.png)
@lipkau  is that what you need?

**Comment by lipkau** [comment]
Almost.
on the last message of the first screenshot, enter `s` and hit enter
then you can see the content of the local variables. such as `$Body`

**Comment by michalporeba** [comment]
I'll do it tomorrow

**Comment by michalporeba** [comment]
I have noticed two more things this morning when running the command with the `-Debug` option. 
First of all it fails only for some of the users, specifically so far I have only noticed that it fails only for my user, for other users I tried it works. 

During execution of the `Set-JiraIssue` a call is made to `rest/api/latest/user/search?` and then 3 different calls to `rest/api/2/user` passing 3 different user names: mine and two other usernames that start with exact my username + and extra character at the end. 

So I tried `Get-JiraUser -UserName <myusername>` and I got the same behaviour - 3 users not 1.

**Comment by lipkau** [comment]
This behavior is know (#49) and already planned to changed (#306).
But as it's a breaking change, it will only be released with v3.0.

for now you will have to manage with something like
```powershell
Get-JiraUser -UserName <myusername> | Where name -eq <myusername>
```

**Comment by michalporeba** [comment]
Please have a look at this merge request. It fixes the issue without breaking changes.

**Comment by michalporeba** [comment]
I have a unit test to cover the `-Exact` option and modified one to be more explicit how the Get-JiraUser should work when the `-Exact` is not used. I have also updated the `Get-JiraUser` documentations.
